### ISSUE TITLE
DeckSwiper doesn't trigger onSwipeLeft

### ISSUE THREAD

**Issue by sandeepskak** [GeekyAnts/NativeBase#357]
Setup:
NativeBase version: 0.5.18
react-native: 0.39.2

Context: 
I used the sample code in the docs of [0.5.7](http://nativebase.io/docs/v0.5.7/components#cardSwiper).
I added logs in the `next()` and `previous()` functions. According to the sample code, on swiping left, the `previous()` function should get called right? Because of `onSwipeLeft={()=>this.previous()`. But this never gets called, my app always loggs the call to `next()`

I did a little digging inside DeckSwiper.js and found that whether i swipe right or left, `vx` always remains 0: 
```
onPanResponderRelease: (e, {vx, vy}) => {
    if(this.props.onSwiping)
      this.props.onSwiping(null);
    var velocity;

    if (vx >= 0) {
        velocity = clamp(vx, 4.5, 10);
    } else if (vx < 0) {
        velocity = clamp(vx * -1, 4.5, 10) * -1;
    }

    if (Math.abs(this.state.pan.x._value) > SWIPE_THRESHOLD) {

        if (velocity>0) {
            (this.props.onSwipeRight) ? this.props.onSwipeRight() : undefined;
            this.selectNext();
        } else {
            (this.props.onSwipeLeft) ? this.props.onSwipeLeft() : undefined;
            this.selectNext();
        }

        Animated.decay(this.state.pan, {
            velocity: {x: velocity, y: vy},
            deceleration: 0.98
        }).start(this._resetState.bind(this))
    } else {
        Animated.spring(this.state.pan, {
            toValue: {x: 0, y: 0},
            friction: 4
        }).start()
    }
}
```

Hence i checkout the docs for [PanResponder](https://facebook.github.io/react-native/docs/panresponder.html) and changed the definition of `onPanResponderRelease` to the following:

```
onPanResponderRelease: (e, {dx, dy}) => {
    if(this.props.onSwiping)
      this.props.onSwiping(null);
    var velocity;

    if (dx >= 0) {
        velocity = clamp(dx, 4.5, 10);
    } else if (dx < 0) {
        velocity = clamp(dx * -1, 4.5, 10) * -1;
    }
    if (Math.abs(this.state.pan.x._value) > SWIPE_THRESHOLD) {

        if (velocity>0) {
            (this.props.onSwipeRight) ? this.props.onSwipeRight() : undefined;
            this.selectNext();
        } else {
            (this.props.onSwipeLeft) ? this.props.onSwipeLeft() : undefined;
            this.selectNext();
        }

        Animated.decay(this.state.pan, {
            velocity: {x: velocity, y: dy},
            deceleration: 0.98
        }).start(this._resetState.bind(this))
    } else {
        Animated.spring(this.state.pan, {
            toValue: {x: 0, y: 0},
            friction: 4
        }).start()
    }
}
```
In short i replaced `vx` and `vy` with `dx` and `dy` instead. For my use case it works, the app triggers `previous()` on swiping left and `next()` on swiping right. But there seems to be flickering of the UI with this change.
Just to make sure, i cloned the repo and fetched the latest master. The code in DeckSwiper.js is still the same as posted above.

**Comment by sankhadeeproy007** [comment]
I can see that `onSwipeRight` and `onSwipeLeft` are both working. Is there anything that I'm missing?
![swiper](https://cloud.githubusercontent.com/assets/10998047/21538584/90b64f1c-cdc4-11e6-9bd7-a70b8f51c67a.gif)

**Comment by gianpaj** [comment]
It worked for me if I did this:
```js
  onSwipeLeft() {
    console.log('left called');
  }

  render() {
    return (
      <Container style={styles.container}>
        <Header>
          <Title>Deck Swiper</Title>

          <Button transparent onPress={this.props.openDrawer}>
            <Icon name="ios-menu" />
          </Button>
        </Header>

        <View padder >
          <DeckSwiper
            dataSource={cards}
            onSwipeLeft={this.onSwipeLeft}
```

Not sure if there's a better way to bind the function from props or otherwise

**Comment by sandeepskak** [comment]
@sankhadeeproy007 @gianpaj i was facing this issue while running AVD(Android Virtual Device) API 23, Android 6.0. Seems to be working fine on iOS as described by @sankhadeeproy007. I'll close this for now and maybe re-open if i can confirm that the issue is only with Android or with the use of AVD.

**Comment by gianpaj** [comment]
I tried on Android 6.0 using Genymotion. worked fine

**Comment by priyarg** [comment]
is it possible to add differnt sounds onswiperight and onswipeleft of card and onload of card
